pkg/cmd/roachtest/roachtestutil/mixedversion/mixedversion_test.go

```diff
@@ -368,3 +369,23 @@ func withTestBuildVersion(v string) func() {
 	clusterupgrade.TestBuildVersion = testBuildVersion
 	return func() { clusterupgrade.TestBuildVersion = nil }
 }
+
+func TestSupportsSkipUpgradeTo(t *testing.T) {
+	expect := func(verStr string, expected bool) {
+		t.Helper()
+		v := clusterupgrade.MustParseVersion(verStr)
+		if r := clusterversion.Latest.ReleaseSeries(); int(r.Major) == v.Major() && int(r.Minor) == v.Minor() {
+			// We have to special case the current series, to allow for bumping the
+			// min supported version separately from the current version.
+			expected = len(clusterversion.SupportedPreviousReleases()) > 1
```

Comment:
>to allow for bumping the min supported version separately from the current version
			
Why do we need to allow this? Is it because when we go to `25.1`, we need to bump `minVersion` twice but `currentVersion` once? Or something else?

Reply:
So right now master is still 24.3 and we are first bumping the min version to 24.2, then we will bump the current version to 25.1, then we will bump the min supported again (to 24.3).  We want to separate these because they tend to be large PRs. Ideally we would bump the version to 25.1 first but that cannot be done until we have an RC build for 24.3 (it's needed as a previous release for upgrade testing).

I will consider changing the runbooks to reorder these steps to avoid this issue, in which case I'll clean this up.

Reply:
Ack, got it. Thanks for explaining 👍